### DISCUSSION TITLE
@urql/svelte query doesn't return a promise

### DISCUSSION THREAD

**Discussion by whittlem** [urql-graphql/urql#2401] [Q&A] [answered]
@urql/svelte query doesn't return a promise. According to forums and the docs there should be a toPromise() method which does not exist for query. I've also read there is a hack to use mutations, which also doesn't work in urql/svelte. The way it's working now is very unpleasant. The query writes to a store and then the app waits for the store to retrieve data. This seems to be a common problem but I've not found an acceptable solution.

urql version & exchanges:

"@urql/svelte": "^1.3.3"
"urql": "^2.2.0"

**Steps to reproduce**

	query(exampleQuery)
	  .then((result) => {
	   console.log(result);
	  });

**Expected behavior**

A query within a function should be able to wait for fetching to be false before returning.

**Actual behavior**

The function returns the query with fetching true, and a promise does not work in any form.

**Comment by kitten** [comment, marked as answer]
I believe you may have gotten two APIs confused. You can find the exact API details on two different docs pages:

- Svelte: https://formidable.com/open-source/urql/docs/api/svelte/#query
- Core: https://formidable.com/open-source/urql/docs/api/core/#clientquery
